lxd-migrate/main_migrate.go

```diff
@@ -248,11 +251,21 @@ func (c *cmdMigrate) runInteractive(server lxd.InstanceServer) (cmdMigrateData,
 
 	config.InstanceArgs = api.InstancesPost{
 		Source: api.InstanceSource{
-			Type: "migration",
-			Mode: "push",
+			Type:              "conversion",
+			Mode:              "push",
+			ConversionOptions: c.flagConversionOpts,
 		},
 	}
 
+	// If server does not support conversion, fallback to migration.
+	// Migration will move the image to the server and import is as
+	// LXD instance. This means that images of different formats,
+	// such as VMDK and QCow2 will not work.
+	if !server.HasExtension("instance_import_conversion") {
+		fmt.Println("Server does not support image conversion. Using normal migration")
```

Comment:
"normal" here is problematic as the new normal will quickly become the conversion mode.

Suggest printing something more useful to the user (who likely has no knowledge or interest in the underlying mechanics of how its sent):

"Server does not support image conversion. Expecting input is in raw format"

Which we could later improve to use basic client side heuristics to detect if the input file looks like its not raw and reject.

Reply:
Very good point. I've fixed that the error message.

Regarding the client side check, we could use `file` to determine the file type of input image.